src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+
+import { Button } from '../Button/Button.tsx';
+
+import styles from './pagination.module.css';
+
+interface PaginationProps {
+    totalItems: number;
+    itemsPerPage: number;
+    onPageChange: (pageNumber: number) => void;
+}
+
+const Pagination = ({ totalItems, itemsPerPage, onPageChange }: PaginationProps) => {
+    const [currentPage, setCurrentPage] = useState(1);
+
+    const pageNumbers = [];
+    for (let index = 1; index <= Math.ceil(totalItems / itemsPerPage); index++) {
+        pageNumbers.push(index);
+    }
+
+    const firstPageIndex = Math.max(currentPage - 1, 1);
+    const lastPageIndex = Math.min(firstPageIndex + 2, Math.ceil(totalItems / itemsPerPage));
+
+    const paginate = (pageNumber: number) => {
+        setCurrentPage(pageNumber);
+        onPageChange(pageNumber);
+    };
+
+    const goToPreviousPage = () => {
+        if (currentPage > 1) {
```

Comment:
try to write condition from negative condition
```suggestion
        if (currentPage <=1) return;
        // otherwise logic 
```